### ISSUE TITLE
Fix Request: AvoidUnconditionalBuiltLogStrings false positive with conditional logging

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#557]
```
import lombok.extern.slf4j.Slf4j;
@Slf4j
public class Foo {
    public void log() {
        if (log.isInfoEnabled()) {
            StringBuilder sb = new StringBuilder();
            sb.append("a"); // <-- false positive
            log.info("{}", sb);
        }    
    }
}
```

**Comment by jborgers** [comment]
The example has the same issue
